sys/net/gnrc/link_layer/lorawan/gnrc_lorawan_mlme.c

```diff
@@ -136,6 +136,7 @@ void gnrc_lorawan_mlme_process_join(gnrc_lorawan_t *mac, gnrc_pktsnip_t *pkt)
     gnrc_lorawan_process_cflist(mac, out + sizeof(lorawan_join_accept_t) - 1);
     mac->mlme.activation = MLME_ACTIVATION_OTAA;
     status = GNRC_LORAWAN_REQ_STATUS_SUCCESS;
+    puts("Successfully joined");
```

Comment:
This is a `sys` component. No puts should be there (otherwise this adds a dependency to stdio and adds ROM when in most cases is not necessary).

If you want to add a debug message, please add a DEBUG and add a prefix (e.g `DEBUG("[gnrc_lorawan]: Successfully joined!");`

Reply:
Thanks for pointing this out. Even if debug is disabled shouldn't there be user feedback for successful join, similar to ones after successful uplinks ( confirmed ones) ?

Reply:
Yes, there should be one. At that time there weren't too many mechanisms to inform an upper layer. I guess there's now the [Message Bus](https://doc.riot-os.org/msg__bus_8h.html). This could be used in a way that an upper layer can "subscribe" to a "Netif Up" event

Reply:
As you had suggested, I shall use a separate PR for the same.

Reply:
Yes. But as discussed offline, that should be a generic GNRC netif component.

Reply:
Yup :+1: